Realm/Realm/Realm.cs

```diff
@@ -1323,16 +1322,28 @@ public void RemoveAll()
         /// non-null <see cref="RealmConfigurationBase.EncryptionKey"/>, the copy will be encrypted with that key.
         /// </summary>
         /// <remarks>
-        /// The destination file cannot already exist.
-        /// <para/>
-        /// If this is called from within a transaction it writes the current data, and not the data as it was when
-        /// the last transaction was committed.
+        /// 1. The destination file cannot already exist.
+        /// 2. When using a local Realm and this is called from within a transaction it writes the current data,
+        ///    and not the data as it was when the last transaction was committed.
+        /// 3. When using Sync, it is required that all local changes are synchronized with the server before the copy can be written.
+        ///    This is to be sure that the file can be used as a starting point for a newly installed application.
+        ///    The function will throw if there are pending uploads.
         /// </remarks>
         /// <param name="config">Configuration, specifying the path and optionally the encryption key for the copy.</param>
         public void WriteCopy(RealmConfigurationBase config)
         {
             Argument.NotNull(config, nameof(config));
 
+            if (config is SyncConfiguration && Config is SyncConfiguration)
+            {
+                SyncConfiguration originalConfig = (SyncConfiguration)Config;
+                SyncConfiguration copiedConfig = (SyncConfiguration)config;
```

Comment:
This can be rewritten using pattern matching `is` expression:

```suggestion
            if (config is SyncConfiguration copiedConfig && Config is SyncConfiguration originalConfig)
            {
```

Reply:
Learned something new today. 👍 
95f232c incorporates that and simplifies the whole check very nicely.